### PR TITLE
Added methods to create a user, and to get all users

### PR DESCRIPTION
**Pull request by Noel-Hann**
You can create a new user by giving it a json input with a username and password with the /api/users/add rout. You can also get all users with the api/users/get-all route.

### CODE REVIEW

**Comment by Bedolla233** [comment]
Before accepting, can you add the screenshots from postman to show that the test are passing.  Use a similar style to what @Jor-Moreno has [here](https://github.com/Noel-Hann/SE_P2_backend/pull/19)

File: src/main/java/com/jorge/whitelist/controller/UserController.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        }
+        User newUser = new User(username, password);
+
+        userRepository.save(newUser);
+        return newUser;
+
+
+    }
+
+//    @GetMapping("/test")
```
Comment (Jor-Moreno) [inline src/main/java/com/jorge/whitelist/controller/UserController.java:48]: nit: We don't want things like this to be in our code since its a bit messy.

**Review by Bedolla233 — APPROVED** [review]
Everything looks good, just make sure to add the Postman test as a comment

**Review by Jor-Moreno — APPROVED** [review]
Accepting, test cases using postman were show in private slack conversation but should be added to PR after the fact.

**Comment by Noel-Hann** [comment]
Screenshot 2024-10-08 231445
Screenshot 2024-10-08 231517
Screenshot 2024-10-08 231547
Screenshot 2024-10-08 231607

**Comment by Noel-Hann** [comment]
These images are my tests.
First image, get-all. There are currently 2 users in the db
Second image, trying to add test2. This returns null because test2 already exists in the db
Third image is adding test3. It returns the values for test3
Fourth image is getting all again. There are now 3 users in the db, including test3 which we added

**Pull request merged**